build-release-tools/lib/PlatformClients.py

```diff
@@ -209,3 +256,93 @@ def upload_a_file(self, package, version, file_path):
         except Exception, ex:
             raise RuntimeError("Failed to upload file {0} due to {1}".format(file_path, ex))
         return True
+
+    def get_package_versions(self, package):
+        """
+        Get the version list of one package.
+        """
+        get_package_versions_url = "/".join([self._api_url, "packages", self._subject, self._repo, package])
+        versions = []
+        resp = self.session.get(get_package_versions_url)
+        if resp.ok:
+            versions = resp.json()["versions"]
+        elif resp.status_code == 404:
+            print "Bintray package: {0}/{1}/{2} doesn't exist".format(self._subject, self._repo, package)
+        else:
+            raise Exception("Failed to get package versions: {0}\n{1}".format(package, resp.text))
+        return versions
+
+    def del_package_version(self, package, version):
+        """
+        Delete a version of one package
+        """
+        del_package_version_url = "/".join([self._api_url, "packages", self._subject, self._repo, package, "versions", version])
+        print del_package_version_url
+        resp = self.session.delete(del_package_version_url)
+        if resp.ok:
+            print "Delete {0}/{1} successfully!".format(package, version)
+        else:
+            raise Exception("Failed to delete {0}/{1}\n {2}!".format(package, version, resp.text))
+
+class Dockerhub(object):
+    """
+    A client class of Dockerhub.
+    """
+    def __init__(self, creds, repo, api_url):
+        assert creds, "creds is None!"
+        assert repo, "repo is None!"
+
+        try:
+            # get from env var
+            self.username, self.password = common.parse_credential_variable(creds)
+        except ValueError as e:
+            # not in env var, parse as user:pass
+            self.username, self.password = creds.split(":")
```

Comment:
I'm not sure, are you sure the order is correct ?
from implicit ENV first, then the explicit parameter?